[Refactor] Health check

Desvelao opened the pull request:
### Description

This PR refactors the health check component with modularized checks.
- Add `appConfig` to Redux store (that is used in the health check)
- The checks starts with `Waiting...` state instead of `Checking...`.
- The checks, which has dependencies, will waiting until they are marked are ready
- When all checks are ready, redirect to `/overview`.

### Checks
- With a clean browser (no cookies), go to the app and all check enabled. It should pass all checks, creating the index pattern for alerts, monitoring, and statistics.
- With a clean browser (no cookies), go to the app, and some checks disabled. It should pass all checks, and redirects when all checks are marked as ready.
- Go to `Settings > Configuration` and change the `pattern` setting, click on `Save changes` button and the `Execute health check` button. The health check should create the new index pattern if doesn't exist previously.

Closes: #2682

frankeros commented on public/components/health-check/components/check-result.test.tsx:
```diff
@@ -0,0 +1,88 @@
+/*
+ * Wazuh app - Check Result Component - Test
```
:clap: :clap:

frankeros commented on public/components/health-check/container/health-check.container.tsx:
```diff
@@ -0,0 +1,187 @@
+  useEffect(() => {
+    // Redirect to app when all checks are ready
+    Object.keys(checks)
+      .every(check => checksReady[check]) && (window.location.href = getHttp().basePath.prepend('/app/wazuh#/overview'));
```
when the health check is done and ok, it maybe would return to the prev location, but it would go to overview if there is the first navigation in the app.

Desvelao replied:
👍

frankeros commented on public/components/health-check/services/check-pattern-support.service.ts:
```diff
@@ -0,0 +1,43 @@
+import { getToasts } from '../../../kibana-services';
+
+export const checkPatternSupportService = async (pattern: string, indexType : string): Promise<{ errors: string[] }> => {
+  let errors: string[] = [];
```
const

frankeros commented on public/components/health-check/services/check-pattern.service.ts:
```diff
@@ -0,0 +1,40 @@
+
+export const checkPatternService = async (appConfig): Promise<{ errors: string[] }> => {
+  const patternId = AppState.getCurrentPattern();
+  let errors: string[] = [];
```
const

frankeros commented on public/components/health-check/services/check-pattern.service.ts:
```diff
@@ -0,0 +1,40 @@
+  const existsDefaultPattern = await SavedObject.existsIndexPattern(appConfig.data['pattern']);    
+  existsDefaultPattern.status && getDataPlugin().indexPatterns.setDefault(appConfig.data['pattern'], true);
+
+  let patternData = patternId ? await SavedObject.existsIndexPattern(patternId) : false;
+
+  if (!patternData) patternData = {};
```
simplify by

```
  const patternData = patternId ? (await SavedObject.existsIndexPattern(patternId)) || {} : {} ;
```

Desvelao replied:
That is better. Thank you! 👍

frankeros commented on public/components/health-check/services/check-setup.service.ts:
```diff
@@ -0,0 +1,44 @@
+import { AppState, GenericRequest, WzRequest } from '../../../react-services';
+
+export const checkSetupService = async (): Promise<{ errors: string[] }> => {
+  let errors: string[] = [];
```
const

frankeros commented on public/components/health-check/services/check-template.service.ts:
```diff
@@ -0,0 +1,30 @@
+import { AppState, GenericRequest, SavedObject } from '../../../react-services';
+
+export const checkTemplateService = async (): Promise<{ errors: string[] }> => {
+  let errors: string[] = [];
```
const

frankeros commented on public/components/health-check/services/check-template.service.ts:
```diff
@@ -0,0 +1,30 @@
+  let errors: string[] = [];
+  const patternId = AppState.getCurrentPattern();
+  let patternTitle = '';
+  let patternData = patternId ? await SavedObject.existsIndexPattern(patternId) : false;
+  if (!patternData) patternData = {};
```
same here

frankeros reviewed (commented):
Amazing job :heart:

Desvelao commented:
### Changes
- Request changes
- Added the max buckets check when the component is mounted
- Created the `useRootScope` hook
- Improved the export in the HOCs and hooks index files
- Removed the `lib` folder
- Removed the `health-check` old component

Desvelao commented:
Should we improve the list of index patterns retrieved to search for a valid to use in the app?

pablomarga reviewed (commented):
When you disable the pattern check the health check failed. It also happened in the old health check.
![Screenshot_20210510_155405](https://user-images.githubusercontent.com/46006326/117670330-13d0b880-b1a8-11eb-85b6-fd17ca95f54d.png)

frankeros reviewed (approved):
LGTM! I'm merging this PR now and I moving the problem commented by Pablo  to a new issue
